### PR TITLE
Add user entity

### PR DESCRIPTION
**Pull request by deyvidholz**
## Migrations

NO

## Description

Add user entity and bcryptjs as dependency to encrypt user's password.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Steps to Test or Reproduce

Follow README.

## Produced Queries

```sql
# there are no new produced or changed queries
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/entities/user.entity.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'users' })
+export class User {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ unique: true })
```
Comment (lucasmonstro) [inline src/entities/user.entity.ts:8]: I suggest adding index to get more performance:

```suggestion
  @Index()
  @Column({ unique: true })
```

File: src/entities/user.entity.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'users' })
```
Comment (lucasmonstro) [inline src/entities/user.entity.ts:3]: I suggest you removing table name, typeorm will handle it for you changing classname to lowercase and adding plural. The result will be `users`.

```suggestion
@Entity()
```

File: src/app.module.ts [hunk 3]
```diff
@@ -16,6 +17,7 @@ import { AppService } from './app.service';
         username: configService.get('POSTGRES_USER'),
         password: configService.get('POSTGRES_PASSWORD'),
         database: configService.get('POSTGRES_DB'),
+        entities: [User],
```
Comment (lucasmonstro) [inline src/app.module.ts:20]: You dont need to load entities here since you have `autoLoadEntities: true`.

**Review by lucasmonstro — CHANGES_REQUESTED** [review]

File: src/entities/user.entity.ts (hunk 1 above)
Comment (deyvidholz) [reply src/entities/user.entity.ts:9]: Resolved on 8568ec2e0a63a0a8a642d4dabda632c40e4a9f2a.

File: src/entities/user.entity.ts (hunk 2 above)
Comment (deyvidholz) [reply src/entities/user.entity.ts]: Resolved on 17dd89ff8123681c362f061a4bb75f94f1f18451.

File: src/app.module.ts (hunk 3 above)
Comment (deyvidholz) [reply src/app.module.ts]: Resolved on 3aa3aae6f20eeaff72b35bbe93b213801d9ce268.

**Review by lucasmonstro — APPROVED** [review]
Looks good for me! Ready for merge!

**Pull request merged**